### PR TITLE
Update  what-is-the-arguments-object.md

### PR DESCRIPTION
**Pull request by Yash-Handa**
Hey @chowdhurian and @fhemberger,
In accordance with issue #1977, I have updated the Article **what-is-the-arguments-object** with the following changes:

* Added details about `arrow function` exception and its workaround (`rest parameter`)
* Added `Array.from()` method usage.
* Added better syntax high lighting.
* some other miscellaneous updates and refactors.

If there are any **suggestion/enhancement/tips** please feel free to tell me, I would love to incorporate those changes in this Article 😄

### CODE REVIEW

**Review by ghost — APPROVED** [review]

File: locale/en/knowledge/javascript-conventions/what-is-the-arguments-object.md [hunk 1]
```diff
@@ -24,41 +24,63 @@ property of `arguments` is callee. This always refers to the function
...
 
-    var myfunc = function(one) {
-      arguments.callee === myfunc;
-      arguments[0] === one;
-      arguments[1] === 2;
-      arguments.length === 3;
-    }
+```js
+const myfunc = function(one) {
+  arguments.callee === myfunc;
```
Comment (fhemberger) [inline locale/en/knowledge/javascript-conventions/what-is-the-arguments-object.md]: `arguments.callee` has been deprecated in ES5 strict mode:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/arguments/callee

File: locale/en/knowledge/javascript-conventions/what-is-the-arguments-object.md [hunk 2]
```diff
@@ -24,41 +24,63 @@ property of `arguments` is callee. This always refers to the function
...
-      args.push(arguments[i]);
-    }
+```js
+//  for ES5 and below
+const args = [];
+for(let i = 0; i < arguments.length; i++) {
+  args.push(arguments[i]);
+}
+
+// for ES6 and above
```
Comment (fhemberger) [inline locale/en/knowledge/javascript-conventions/what-is-the-arguments-object.md]: Maybe we should push this as the default for current Node.js versions (and only mention the pre-ES6 variants briefly). What do you think?

**Review by fhemberger — CHANGES_REQUESTED** [review]

File: locale/en/knowledge/javascript-conventions/what-is-the-arguments-object.md (hunk 2 above)
Comment (Yash-Handa) [reply locale/en/knowledge/javascript-conventions/what-is-the-arguments-object.md]: Thanks, @fhemberger for the advice will update both the sections :)

**Review by fhemberger — APPROVED** [review]

**Comment by fhemberger** [comment]
Thanks!

**Pull request merged**